src/components/protocol_handler/test/protocol_handler_tm_test.cc

```diff
@@ -3367,11 +3482,872 @@ TEST_F(ProtocolHandlerImplTest, SendMessageToMobileApp_SendMultiframeMessage) {
+  // In this call we cannot track any calls and values
+  // So there is just covering of lines
+  auto protocol_handler_implas_listener =
+      static_cast<TransportManagerListener*>(protocol_handler_impl.get());
+  protocol_handler_implas_listener->OnConnectionClosed(connection_id);
+}
+
+TEST_F(ProtocolHandlerImplTest, SendFrame_NullMessage_ResultFail) {
+  auto protocol_handler_implas_listener =
+      static_cast<protocol_handler::impl::ToMobileQueue::Handler*>(
+          protocol_handler_impl.get());
+  EXPECT_CALL(transport_manager_mock, SendMessageToDevice(_)).Times(0);
+  protocol_handler::impl::RawFordMessageToMobile null_message;
+  protocol_handler_implas_listener->Handle(null_message);
+}
+
+TEST_F(ProtocolHandlerImplTest,
+       SendFrame_SendMessageToDeviceFailed_ResultFail) {
+  auto protocol_handler_implas_listener =
+      static_cast<protocol_handler::impl::ToMobileQueue::Handler*>(
+          protocol_handler_impl.get());
+  protocol_handler::impl::RawFordMessageToMobile message(
+      std::make_shared<protocol_handler::ProtocolPacket>(connection_id,
+                                                         PROTOCOL_VERSION_3,
+                                                         PROTECTION_OFF,
+                                                         FRAME_TYPE_CONTROL,
+                                                         kControl,
+                                                         FRAME_DATA_HEART_BEAT,
+                                                         session_id,
+                                                         some_data.size(),
+                                                         message_id,
+                                                         &some_data[0]),
+      kFinalMessage);
+  EXPECT_CALL(transport_manager_mock, SendMessageToDevice(_))
+      .WillOnce(Return(transport_manager::E_INTERNAL_ERROR));
+  protocol_handler_implas_listener->Handle(message);
+}
+
+TEST_F(ProtocolHandlerImplTest, HandleSingleFrameMessage_ResultOk) {
+  auto protocol_handler_implas_listener =
+      static_cast<protocol_handler::impl::FromMobileQueue::Handler*>(
+          protocol_handler_impl.get());
+  protocol_handler::impl::RawFordMessageFromMobile message(
+      std::make_shared<protocol_handler::ProtocolPacket>(connection_id,
+                                                         PROTOCOL_VERSION_3,
+                                                         PROTECTION_OFF,
+                                                         FRAME_TYPE_SINGLE,
+                                                         kControl,
+                                                         FRAME_DATA_SINGLE,
+                                                         session_id,
+                                                         1u,
+                                                         message_id,
+                                                         &some_data[0]));
```

Comment:
Unclear meaning of pointer to "some data" with index, maybe it will be better to create constant or variable with self-explained name.

Reply:
Please check all similar cases across the file